paper.md

```diff
@@ -2,11 +2,88 @@
+they solve most of the problems caused by using VM's. So, people started replacing VM's with containers in workflow engines for providing platform independent reproducibility [@repocompscience2016][@piccolo2016tools]. Since, these container runtimes are available for almost every well known operating systems and architectures, experiments can be reproduced easily using containerized workflows in almost any environment [@zheng_integrating_2015].
+
+<!-- this is how people tried solving reproducibility with containers -->
+
+Currently, the different container based workflow execution engines that are available can be classified mainly into two categories. First category involves engines that are inherently container-native but assume the presence of a fully provisioned Kubernetes [@kubernetes_google] cluster at their disposal for workflow execution. Therefore, this category of workflow engines could also be termed as cloud-native [@balalaie2016microservices]. Some popular examples of this type of workflow execution engines are Argo [@argocommunity_argoproj_2019], Pachyderm[@novella_containerbased_2018] and Brigade[@brigade]. The other category of workflow engines do not enforce container based execution by default but provides some plugins and extensions to execute steps in containers. Nextflow [@ditommaso_nextflow_2017] and Pegasus [@deelman_pegasus_2004] are some popular examples of this category.
+
+<!-- this is what the above problems don't solve -->
+
+After studying the different workflow execution engines available, we discovered
+the absence of a possible third category, which should enforce the execution of workflows inside containers, hence making it container-native but should not enforce the presence of a Kubernetes cluster or a cloud computing environment. The presence of a Kubernetes cluster or a cloud computing environment should not be a hard core requirement for reproducing any experiment since it is often costly [@rodriguez2020container] to get access to one and inturn makes reproducibilty complex. Hence, it should provide flexibility for their users to run workflows in a wide range of computing environments like a single node local machine, a HPC environment [@yang2005high], a Kubernetes cluster or any cloud computing environment of their choice.
```

Comment:
the first two sentences should be clear to the reader after reading the part where we mention container technology. It's OK to explicitly state it, but if we did a good job, this should be evident.

The third (last sentence) can be used to start talking about our contributions

Reply:
@ivotron Can you please tell me what changes I should make in the first two sentences?

Reply:
Sorry for the confusion. What I meant is that, the part that explains VMs/containers, it will also explain that there is no way of running a workflows locally, that they can only run in kubernetes. Then, the part that describes workflow engines will also discuss that there are not container-native ones. So, if these two parts do a good job of explaining this lack on functionality, it shouldn't be necessary to explicitly say "After studying the different...."

In other words, what I'm saying is that this paragraph merged with the previous parts that talk about containers/workflows. Does that make sense?

Reply:
Yeah. understood. Thank you